aya/src/util.rs

```diff
@@ -200,9 +201,25 @@ pub fn online_cpus() -> Result<Vec<u32>, io::Error> {
 
 /// Get the number of possible cpus.
 ///
+/// Caches the value into thread local value on the first successful access
 /// See `/sys/devices/system/cpu/possible`.
 pub fn nr_cpus() -> Result<usize, io::Error> {
-    Ok(possible_cpus()?.len())
```

Comment:
Why not cache in `possible_cpus`? Surely the most expensive thing is file system interaction.

Reply:
It's used only inside nr_cpus, but I can change it to 

```rs
fn possible_cpus() -> Result<&'static [u32], io::Error>
```
Cloning vector seems costly

Reply:
Yes, changing the signature is fine.

Reply:
> Yes, changing the signature is fine.

```rust
pub(crate) fn possible_cpus() -> Result<&'static [u32], io::Error> {
    thread_local! {
        static CPUS_LIST: RefCell<Option<Vec<u32>>> = const {RefCell::new(None)};
    }

    CPUS_LIST.with(|cell| {
        if let Some(cpus) = &*cell.borrow() {
            Ok(cpus.as_slice())
        } else {
            let data = fs::read_to_string(POSSIBLE_CPUS)?;
            let cpus = parse_cpu_ranges(data.trim()).map_err(|_| {
                io::Error::new(
                    io::ErrorKind::Other,
                    format!("unexpected {POSSIBLE_CPUS} format"),
                )
            })?;

            *cell.borrow_mut() = Some(cpus.clone());

            Ok(cpus.as_slice())
        }
    })
}
```

It won't compile :(
![image](https://github.com/user-attachments/assets/99d4c40c-cf24-4e0c-b2aa-8d40bd27b286)

So we should either use `OnceLock` or just cache `u32`. I'm voting for 2 variant

Reply:
I sent https://github.com/aya-rs/aya/pull/1045 in which I remove `possible_cpus`. This can then cache the u32 without seeming strange.

Reply:
@tamird should I rebase on it?

Reply:
Yep, looks good to me :)

> Rebased this, does it look reasonable to you?